ipaserver/install/server/upgrade.py

```diff
@@ -1380,22 +1349,56 @@ def fix_trust_flags():
     sysupgrade.set_upgrade_state('http', 'fix_trust_flags', True)
 
 
-def export_ra_agent_pem():
-    root_logger.info('[Exporting KRA agent PEM file]')
+def update_ra_agent_crt():
 
-    # export_kra_agent_pem is the original name of this function
-    sysupgrade.remove_upgrade_state('http', 'export_kra_agent_pem')
+    root_logger.info('[Exporting RA agent PEM files]')
+    _fd, filename = tempfile.mkstemp(dir=paths.VAR_LIB_IPA)
 
     if os.path.exists(paths.RA_AGENT_PEM):
-        root_logger.info("KRA agent PEM file already exported")
-        return
-
-    if not api.Command.kra_is_enabled()['result']:
-        root_logger.info("KRA is not enabled")
+        root_logger.info("RA agent PEM files already exported")
         return
 
-    dogtaginstance.export_ra_agent_pem()
-    installutils.remove_file(paths.OLD_KRA_AGENT_PEM)
+    ra_nick = 'ipaCert'
+    args = [paths.PKI,
+            "-d", paths.HTTPD_ALIAS_DIR,
+            "-C", os.path.join(paths.HTTPD_ALIAS_DIR, 'pwdfile.txt'),
+            "client-cert-show", ra_nick,
+            "--client-cert", filename]
+    ipautil.run(args)
+
+    # get the private key and certificate from the file
+    with open(filename, 'r') as f:
+        data = f.read()
+        cert = crypto_x509.load_pem_x509_certificate(
+            data, default_backend())
+        key = serialization.load_pem_private_key(
+            data, password=None, backend=default_backend())
+
+    os.remove(filename)
+
+    with open(paths.RA_AGENT_PEM, 'w') as f:
+        f.write(cert.public_bytes(serialization.Encoding.PEM))
+
+    with open(paths.RA_AGENT_KEY, 'w') as f:
+        f.write(key.private_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.PKCS8,
+            encryption_algorithm=serialization.NoEncryption()
+        ))
+
+    pent = pwd.getpwnam(constants.HTTPD_USER)
+    for pem_file in (paths.RA_AGENT_PEM, paths.RA_AGENT_KEY):
+        os.chown(pem_file, 0, pent.pw_gid)
+        os.chmod(pem_file, 0o440)
+
+    if os.path.exists('/etc/httpd/alias/kra-agent.pem'):
+        os.remove('/etc/httpd/alias/kra-agent.pem')
+
+    httpdb = certs.CertDB(api.env.realm)
+    oldcert = httpdb.get_cert_from_db(ra_nick)
+    if oldcert:
+        certmonger.stop_tracking(paths.HTTPD_ALIAS_DIR, nickname=ra_nick)
+        httpdb.delete_cert(ra_nick)
```

Comment:
This sort of duplicates the `update_ra_cert_store` plugin's functionality, doesn't it?